### DISCUSSION TITLE
Elasticsearch is failing to connect

### DISCUSSION THREAD

**Discussion by Jcremer11** [Security-Onion-Solutions/securityonion#2810] [Unsupported Versions]
Brand new install of SO2 on ESXi 6.5 host. 
VM has 16GB of RAM and 6CPU Cores. (That's all we can spend at the moment)
Install type: Manager & Search Server

When I checked the Elastic logs, it shows the following error: Java.lang.illegalArgumentEception: node with [discovery.type] set to [single-node] must be master-eligible

When I checked the setup in: /opt/so/saltstack/local/pillar/minions/$minion.sls...I see the following:

elasticsearch:
  mainip: 10.0.1.31
  mainint: 'ens192'
  esheap: 4066m
  esclustername: {{ grains.host }}
  node_type: ''
  es_port: 9200
  log_size_limit: 369
  node_route_type: 'hot'

Do I need to change "node_type:" to something else? What am I missing?

**Comment by TOoSmOotH** [comment]
Did you select advanced and true cluster or just use defaults?

**Reply by Jcremer11** [reply]
I selected the defaults. Can you elaborate on what I need to do in this case? 


NOTE: Whenever I perform the install with the defaults, the install finishes and said it "had a problem" and to check the /root/sosetup.log for the errors. Then, when I check the sosetup.log file, it shows: 
"[ERROR] The Salt Master has cached the public key for this node, 
this salt minion will wait for 10 seconds before attempting to re-authenticate. 
Minion failed to authenticate with the master, has the minion key been accepted? 
Restarting service salt-master
Checking service salt-master status
salt-master is running
Checking if we can talk to the salt master
could not talk to salt master
salt minion could not talk to salt master after 30 attempts, exiting
Errors detected during setup: skipping post-setup steps to allow for analysis of failures"

**Reply by Jcremer11** [reply]
...Here's a better explanation of what I'm doing:

I'm trying to install my Manager&Search node on ESXi 6.5
I've given the Manager/Search node 6CPU cores, 16GB of RAM, 1TB of HDD space, and 1 network interface. We will add more storage later as needed. Our retention needs are not great right now. I'm installing from the SO ISO image (which I did verify the signatures) 

However, the installation process keeps failing for some reason (see previous reply), and I get dumped back at the Centos login and I seem to have no sudo rights or anything, and none of the normal so commands are recognized. 

During installation - I choose the advanced options, and select following:
- Deployment type: Distributed 
- Node Type: ManagerSearch
- Zeek for metadata
- Static IP and DNS
- Custom hostname 
- Keep the default docker IP range
- IP to access Web Interface
- Manager is proxy for updates
- ES Heap size: 4066m
- Logstash input thread: 1
- ES Indices closed after: 30 days
*** I leave the remaining settings at their defaults. 
*** When I get to the components section, it asks which of the following I would like to install:
-GRAFANA
-OSQUERY
-WAZUH
-THEHIVE
-PLAYBOOK
-STRELKA
*** I left all of these selected, since I would like to get familiar with all of these tools.
When the installation finishes, I get the error I described above.
